File: leetcode/lc-us/cpp/452.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    int findMinArrowShots(vector<vector<int>> &points) {
        sort(points.begin(), points.end());
        int arrows = 0;
        int n = (int) points.size();
        for (int i = 0; i < n;) {
            arrows++;
            int end = points[i][1];
            int j = i + 1;
            for (; j < n; j++) {
                end = min(end, points[j][1]);
                if (points[j][0] > end) {
                    break;
                }
            }
            i = j;
        }
        return arrows;
    }
};

File: leetcode/lc-us/py/452.py
class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        points.sort()
        res = 0
        i, n = 0, len(points)
        while i < n:
            res += 1
            j = i + 1
            end = points[i][1]
            while j < n:
                if points[j][0] <= end:
                    end = min(end, points[j][1])
                else:
                    break
                j += 1
            i = j
        return res
